crates/polars-core/src/series/mod.rs

```diff
@@ -189,6 +189,9 @@ impl Series {
     }
 
     pub fn is_sorted_flag(&self) -> IsSorted {
+        if self.len() <= 1 {
+            return IsSorted::Ascending;
```

Comment:
Technically it is sorted both Ascending and Descending at the same time 🤔

Reply:
@stinodego Yes I've brought up this limitation of the current sorted flag to Ritchie before. You can also run into this for larger arrays, `[0, 0, 0, 0, 0]` is sorted both ascending and descending.

Reply:
Why do we replace this from `compute_len` to here?

Reply:
Expressions don't necessarily go down that path, e.g. the original implementation of casting `Datetime` to `Time` didn't: https://github.com/pola-rs/polars/pull/13812